test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs

```diff
@@ -162,32 +124,74 @@ private static IPredictor TrainSentimentCore()
                 // Train
                 var trainer = new SdcaMultiClassTrainer(env, new SdcaMultiClassTrainer.Arguments() { MaxIterations = 20 });
                 var trainRoles = new RoleMappedData(trans, label: "Label", feature: "Features");
-                return trainer.Train(trainRoles);
+
+                var predicted = trainer.Train(trainRoles);
+                _consumer.Consume(predicted); 
             }
         }
 
-        public class IrisData
+        [GlobalSetup(Targets = new string[] { nameof(PredictIris), nameof(PredictIrisBatchOf1), nameof(PredictIrisBatchOf2), nameof(PredictIrisBatchOf5) })]
+        public void SetupPredictBenchmarks()
         {
-            [Column("0")]
-            public float Label;
+            _trainedModel = Train(_dataPath);
+            _consumer.Consume(_trainedModel.Predict(_example));
 
-            [Column("1")]
-            public float SepalLength;
+            var testData = new Data.TextLoader(_dataPath).CreateFrom<IrisData>(useHeader: true);
+            var evaluator = new ClassificationEvaluator();
+            _metrics = evaluator.Evaluate(_trainedModel, testData);
+
+            _batches = new IrisData[_batchSizes.Length][];
+            for (int i = 0; i < _batches.Length; i++)
+            {
+                var batch = new IrisData[_batchSizes[i]];
+                _batches[i] = batch;
+                for (int bi = 0; bi < batch.Length; bi++)
+                {
+                    batch[bi] = _example;
+                }
```

Comment:
Does this for loop change elements of `_batches[i]` or only elements of the local variable `batch`?  Not an expert so not sure whether `batch` is a ref-type.

Reply:
this is done on purpose, it's a Setup method